library/ssl_msg.c

```diff
@@ -4325,31 +4325,31 @@ static int ssl_buffer_message( mbedtls_ssl_context *ssl )
                     {
                         /* If we can't buffer a future message because
                          * of space limitations -- ignore. */
-                        MBEDTLS_SSL_DEBUG_MSG( 2, ( "Buffering of future message of size %u would exceed the compile-time limit %u (already %u bytes buffered) -- ignore\n",
-                             (unsigned) msg_len, MBEDTLS_SSL_DTLS_MAX_BUFFERING,
-                             (unsigned) hs->buffering.total_bytes_buffered ) );
+                        MBEDTLS_SSL_DEBUG_MSG( 2, ( "Buffering of future message of size %zu would exceed the compile-time limit %d (already %zu bytes buffered) -- ignore\n",
+                             msg_len, MBEDTLS_SSL_DTLS_MAX_BUFFERING,
```

Comment:
`MBEDTLS_SSL_DTLS_MAX_BUFFERING` could be defined as `1234L` or `0x123456789a`, so we don't know whether it fits `int`. It should be cast to a `size_t`.

Applies everywhere `MBEDTLS_SSL_DTLS_MAX_BUFFERING` is printed out.